js/csv.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Converts a JSON object to a valid CSV string file.
+ *
+ * @param {Object} object The object to convert.
+ * @returns {String} The converted string.
+ */
+export const objectToCsv = object => {
+    let csv = "";
+    if (Object.keys(object) === 0) return csv;
+
+    const fields = Object.keys(object[0]);
+    csv += fields.toString() + "\n";
+
+    for (let i = 0; i < Object.keys(object).length; i++) {
+        let entry = "";
+        for (let j = 0; j < fields.length; j++) {
+            // values containing the ',' character need to be parsed,
+            // otherwise values will not align with fields
+            const value = object[i][fields[j]];
+            if (String(value).includes(",")) {
+                entry += '"' + String(value).replace(/["']/g, "") + '",';
+            } else {
+                // only append truish values
+                entry += value ? value + "," : ",";
+            }
+        }
+        // remove unnecessary final comma
+        csv += entry.slice(0, -1) + "\n";
+    }
+
+    return csv;
+};
+
+/**
+ * Reads a CSV file and returns a table with the data.
+ *
+ * @param {Blob|File} file The file object to be read.
+ * @param {Function} parser The function to be used in the
+ * CSV parsing operation.
+ */
+export const readCsv = (file, parser = null) => {
+    parser = parser || parseCsvComplex;
+    return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.onload = e => resolve(parser(e.target.result));
+        reader.onerror = e => reject(e);
+        reader.readAsText(file);
+    });
+};
+
+/**
+ * Creates the contents of the CSV file with the data given
+ * as an array of rows and the headers.
+ *
+ * @param {Array} data Array of rows with cell data.
+ * @param {Array} headers List of headers for the CSV file.
+ */
+export const buildCsv = (data, headers = []) => {

```

Comment:
This function takes an array of data and converts it to a valid CSV string. Your `objectToCsv` function takes an object and converts it to a valid CSV string.

I would maybe make `buildCsv` check the type of the first parameter: 
- if it is an array, do the logic implemented
- if it is an object, do your logic

Reply:
Upon further testing, the new function for building CSVs which includes objects also work correctly for the previous existing use cases, so I changed the function to only include the new function.